src/controller/deletecontent.js

```diff
@@ -208,9 +215,41 @@ function shouldEntireContentBeReplacedWithParagraph( schema, selection ) {
 		return false;
 	}
 
-	if ( !schema.check( { name: 'paragraph', inside: limitElement.name } ) ) {
-		return false;
-	}
+	return schema.check( { name: 'paragraph', inside: limitElement.name } );
+}
 
-	return true;
+// Gets a name under which we should check this node in the schema.
+//
+// @param {module:engine/model/node~Node} node The node.
+// @returns {String} node name.
+function getNodeSchemaName( node ) {
+	return node.is( 'text' ) ? '$text' : node.name;
+}
+
+// Adds deltas with `removeAttributes` operation for disallowed by schema attributes on given nodes and its children.
```

Comment:
Creates `AttributeDelta`s that removes attributes that are disallowed by schema on given node and its children.